providers/softlayer/src/main/java/org/jclouds/softlayer/compute/strategy/SoftLayerComputeServiceAdapter.java

```diff
@@ -213,6 +218,15 @@ public SoftLayerComputeServiceAdapter(SoftLayerApi api,
          api.getVirtualGuestApi().setTags(result.getId(), templateOptions.getTags());
       }
 
+      Map<String, String> meta = templateOptions.getUserMetadata();
+      if (meta != null) {
+         String notes = meta.get(USER_META_NOTES);
+         if (!Strings.isNullOrEmpty(notes)) {
+            checkArgument(notes.length() <= USER_META_NOTES_MAX_LENGTH, "'notes' property in user metadata should be long at most " + USER_META_NOTES_MAX_LENGTH + " characters.");
```

Comment:
Failing at this point may leak the virtual guest that is created a few lines before. You'd better validate this at the beginning of the method to fail fast and avoid creating unnecessary resources.